cmd/run_controller/main.go

```diff
@@ -39,15 +38,17 @@ func main() {
 	// creates the in-cluster config
 	var config *rest.Config
 	var err error
+	defer klog.Flush()
+	
 	if kubeconfig == "" {
-		log.Printf("In cluster")
+		klog.Infof("In cluster")
```

Comment:
Did someone figure out how [klog.Infof()](https://github.com/kubernetes/klog/blob/master/klog.go#L1379) relates to [klog.V(x).Infof()](https://github.com/kubernetes/klog/blob/master/klog.go#L1330)?
For me it looks like `klog.Infof()` always logs, independent of any configured level - which is ok, but we have to be aware that the "info" is not relate to the log level "info" in other logging libraries.

Reply:
I think the difference between both is [configurable verbosity level](https://github.com/kubernetes/klog/blob/master/klog.go#L1254). In [klog.Infof()](https://github.com/kubernetes/klog/blob/master/klog.go#L1379) verbosity level is set default to _0_ - https://github.com/kubernetes/klog/blob/master/klog.go#L407

Reply:
Logging the severity is based on maximum severity configured, that means all severity levels below the configured severity are logged.
| Configured Level | Possible Logging Levels |
|---|---|
| FATAL | info, warning, error, fatal |
| ERROR | info, warning, error |
| WARNING | info, warning |
| INFO | info |

Reply:
Sure, like in other log frameworks... Did you figure out how to configure those log levels (in contrast to the the v-leveled logging controlled via `-v`)?

Reply:
I think with klog severities are not the basis for log output filtering, but only the orthogonal verbosity level.

Reply:
So we could use both `-v` and `-stderrthreshold`? Sounds like we should decide for one way. Meaning...
- either `klog.Info`, `klog.Warning`, `klog.Error`, ... (logging everything with `V(0)`). We would have to use `-stderrthreshold` to influence verbosity then.
- or `klog.V(2).Info`, `klog.V(1).Info`, `klog.V(0).Error` with `-v`.

What do you think?

Reply:
I would follow the approach of [github.com/go-logr/logr](https://github.com/go-logr/logr):

- "If your code receives an error from a subordinate function call and is logging that error _and not returning it_, use error logs."
- In all other cases log as info with certain level. This implies not to use any other severity levels like warning, panic, debug or trace.